### PR TITLE
APB-4339: Add CGT Support

### PR DESCRIPTION
**Pull request by JTY009**

### CODE REVIEW

File: it/uk/gov/hmrc/agentclientauthorisation/controllers/AgentCreateInvitationISpec.scala [hunk 1]
```diff
@@ -115,6 +115,27 @@ class AgentCreateInvitationISpec extends BaseISpec {
...
+        givenAuthorisedAsAgent(arn)
+        givenGetAgencyNameViaClientStub(arn)
+        givenGetAgencyEmailAgentStub(arn)
+        //TODO Add Known Fact Stub to GET Name - APB-4403
+
+        val requestBody = Json.parse(
+          s"""
+             |{
+             |  "service":"HMRC-CGT-PD",
+             |  "clientIdType":"cgt",
```
Comment (sureshhmrc) [inline it/uk/gov/hmrc/agentclientauthorisation/controllers/AgentCreateInvitationISpec.scala]: can we rename this `"cgt"` to `cgtRef`, 
are these changes working fine end to end ?

File: it/uk/gov/hmrc/agentclientauthorisation/support/TestDataSupport.scala [hunk 2]
```diff
@@ -16,6 +16,7 @@ trait TestDataSupport {
   val serviceITSA = "HMRC-MTD-IT"
   val servicePIR = "PERSONAL-INCOME-RECORD"
   val serviceVAT = "HMRC-MTD-VAT"
+  val serviceCGT = "HMRC-CGT-PD"
```
Comment (sureshhmrc) [inline it/uk/gov/hmrc/agentclientauthorisation/support/TestDataSupport.scala]: i think we are not using it anywhere?

File: app/uk/gov/hmrc/agentclientauthorisation/model/Service.scala [hunk 3]
```diff
@@ -45,6 +45,8 @@ object Service {
 
   val HMRCTERSORG = "HMRC-TERS-ORG"
 
+  val HMRCCGTPD = "HMRC-CGT-PD"
```
Comment (sureshhmrc) [inline app/uk/gov/hmrc/agentclientauthorisation/model/Service.scala]: please update README too ...
incase of trusts we have `|For Trust|HMRC-TRUST-ORG|`

Comment (sureshhmrc) [reply app/uk/gov/hmrc/agentclientauthorisation/model/Service.scala]: do we need any CgtRefBinder ?

Comment (sureshhmrc) [reply app/uk/gov/hmrc/agentclientauthorisation/model/Service.scala]: i think we need to update `AgencyInvitationsController .addLinkToInvitation` to support cgt ?

Comment (sureshhmrc) [reply app/uk/gov/hmrc/agentclientauthorisation/model/Service.scala]: please follow this and see if we can add any more for cgt 
https://github.com/hmrc/agent-client-authorisation/commit/de334d3f3c734f7c7b406447dd20b78457d04d84

Comment (JTY009) [reply app/uk/gov/hmrc/agentclientauthorisation/model/Service.scala]: clientActionUrl is for API. This does not apply.

File: it/uk/gov/hmrc/agentclientauthorisation/controllers/AgentCreateInvitationISpec.scala (hunk 1 above)
Comment (JTY009) [reply it/uk/gov/hmrc/agentclientauthorisation/controllers/AgentCreateInvitationISpec.scala]: ok and yes

**Pull request merged**